packages/gitbook/src/components/TableOfContents/useScrollToActiveTOCItem.tsx

```diff
@@ -0,0 +1,84 @@
+'use client';
+
+import React from 'react';
+
+import { tcls } from '@/lib/tailwind';
+
+const TOCScrollContainerContext = React.createContext<React.RefObject<HTMLDivElement> | null>(null);
+
+export function TOCScrollContainerProvider(
+    props: React.PropsWithChildren<{ withHeaderOffset: boolean; withTrademarkEnabled: boolean }>,
+) {
+    const { withHeaderOffset, withTrademarkEnabled, children } = props;
+    const scrollContainerRef = React.createRef<HTMLDivElement>();
+
+    return (
+        <TOCScrollContainerContext.Provider value={scrollContainerRef}>
+            <div
+                ref={scrollContainerRef}
+                className={tcls(
+                    withHeaderOffset ? 'pt-4' : ['pt-4', 'lg:pt-0'],
+                    'hidden',
+                    'lg:flex',
+                    'flex-grow',
+                    'flex-col',
+                    'overflow-y-auto',
+                    'lg:gutter-stable',
+                    'lg:pr-2',
+                    'group-hover:[&::-webkit-scrollbar]:bg-dark/1',
+                    'group-hover:[&::-webkit-scrollbar-thumb]:bg-dark/3',
+                    '[&::-webkit-scrollbar]:bg-transparent',
+                    '[&::-webkit-scrollbar-thumb]:bg-transparent',
+                    'dark:[&::-webkit-scrollbar]:bg-transparent',
+                    'dark:[&::-webkit-scrollbar-thumb]:bg-transparent',
+                    'dark:group-hover:[&::-webkit-scrollbar]:bg-light/1',
+                    'dark:group-hover:[&::-webkit-scrollbar-thumb]:bg-light/3',
+                    'navigation-open:flex', // can be auto height animated as such https://stackoverflow.com/a/76944290
+                    'lg:-ml-5',
+                    withTrademarkEnabled ? 'lg:pb-20' : 'lg:pb-4',
+                )}
+            >
+                {children}
+            </div>
+        </TOCScrollContainerContext.Provider>
+    );
+}
+
+// Offset to scroll the table of contents item by.
+const TOC_ITEM_OFFSET = 200;
```

Comment:
What does this offset represent? why 200?

Reply:
It's an offset to position the scrolled item away from the very top of the scroll container. I choose 200 abritrarily.